docs/user_guides/fs/feature_view/helper-columns.md

```diff
@@ -0,0 +1,127 @@
+# Helper columns
+
+HSFS provides functionality to define two types of helper columns `inference_helper_columns` and `training_helper_columns` to [feature views](./overview.md).
+
+`inference_helper_columns` are a list of feature names that are not used in training the model itself but are used to compute features during batch or online inference. For example in credit card fraud detection system 
+features like days until credit card expires `days_until_card_expires` and distance between previous and current place of transaction `loc_delta_t_minus_1` need to be computed when new transaction arrives. Feature `days_until_card_expires` will be computed
+using credit card expiry date `expiry_date` that is stored in credit card profile feature group and needs to be fetched from the feature store, as well as date of transaction `date_of_transaction` that arrives at request time. To compute `loc_delta_t_minus_1`
+previous transaction coordinates `longitude` and `latitude` needs to fetched from the feature store and compared to new transaction coordinates that arrives at inference application. In this use case `expiry_date`, `date_of_transaction`, `longitude` and `latitude`
+are `inference_helper_columns`. They are not used for training but are necessary for computing on-demand/derived features.
+
+`training_helper_columns` are a list of feature names that are not the part of the model schema itself but are used during training for the extra information. For example one might want to use feature like `category` of the purchased 
+product to assign different weights during the training time.
+
+## Definition
+Both inference and training helper column name(s) must be part of the `Query` object. If helper column name(s) belong to feature group that is part of a `Join` with `prefix` defined, then this prefix needs to prepended
+to the original column name when defining helper column list.
+
+=== "Python"
+
+    !!! example "Define inference and training helper columns with feature views."
+        ```python
+        # define query object 
+        query = label_fg.select("fraud_label")\
+                        .join(cc_profile.select("expiry_date"))\
+                        .join(trans_fg.select(["category", "amount",  "days_until_card_expires", "date_of_transaction"
+                                               "locaton_delta", "longitude", "latitude", "category"])) \
+                        .join(window_aggs_fg.select_except(["trans_volume_mstd", "trans_volume_mavg", "trans_freq", 
```

Comment:
Do we need to have some many feature groups here? It's not htat people can copy paste it directly, so I think we should limit the setup code to the bare minimum and focus on the important part, the feature view definition.